### PR TITLE
Radio bridge download MakeCode hex flow

### PR DESCRIPTION
**Pull request by microbit-grace**
https://microbit-global.monday.com/boards/1550536443/pulses/1588641704

### CODE REVIEW

File: src/get-next-connection-state.ts [hunk 1]
```diff
@@ -74,6 +75,7 @@ export const getNextConnectionState = ({
       currStatus === ConnectionStatus.Connected
         ? ConnectionStatus.ConnectionLost
         : ConnectionStatus.FailedToReconnect;
+
```
Comment (microbit-grace) [inline src/get-next-connection-state.ts:78]:
```suggestion
```

**Comment by microbit-matt-hillsdon** [comment]
Pushed merge of react into this branch to fix trivial strings conflict.

**Comment by microbit-robert** [comment]
We don't remember that you previously connected via radio bridge, so when you attempt to connect for a second time after flashing a MakeCode project, you need to select the bridge option again. Maybe we can remember the user's choice in a session?

**Comment by microbit-grace** [comment]
> Should this still be the first dialog in the flow when using radio bridge? It feels much less clear when there are already two micro:bits in play compared to the single micro:bit with Web Bluetooth approach? 

Hmmm 🤔 Yea, maybe an alternative version of the dialog should be used for the radio case. Will work on suggesting an alternative.

**Comment by microbit-grace** [comment]
> We don't remember that you previously connected via radio bridge, so when you attempt to connect for a second time after flashing a MakeCode project, you need to select the bridge option again. Maybe we can remember the user's choice in a session?

[5f4d608](https://github.com/microbit-foundation/ml-trainer/pull/337/commits/5f4d608c33a91fe342d9f343580183107df7be63)

File: lang/ui.en.json [hunk 2]
```diff
@@ -994,5 +1018,17 @@
   "unknown": {
     "defaultMessage": "unknown",
     "description": "Label for unknown ML event"
+  },
+  "unplug-radio-link-microbit-description": {
+    "defaultMessage": "Unplug the radio link micro:bit from the computer. If you want to record more data samples, you will need to reconnect with the tool.",
```
Comment (microbit-matt-hillsdon) [inline lang/ui.en.json:1023]: Maybe use consistent copy with the BT scenario?

```suggestion
    "defaultMessage": "Unplug the radio link micro:bit from the computer. You will need to reconnect the micro:bit if you want to record more data samples.",
```

**Comment by microbit-grace** [comment]
> We can probably tell if the user is about the flash the wrong (link/bridge) micro:bit before we actually do it. We could add an extra warning step here, but perhaps it's more effort that it's worth.

Will do this in a separate PR, I assume this would be good for pre-beta?

**Pull request merged**